### PR TITLE
Integration tests for session recipe with jwt enabled

### PR DESCRIPTION
**Pull request by nkshah2**
## Summary of change
- Adds tests for using the Session recipe with jwt enabled.
- Adds tests for both axios and fetch
- Adds a new API `/featureFlags GET` to the test server to be able to check SDK version for different features. In this PR it is used to check if the Node SDK version is greater than equal to 8.3
- Changes the pre commit script to check for tests that use `.only`

## Related issues
- https://app.zenhub.com/workspaces/supertokens-60df058bb63a6a0015cdd807/issues/supertokens/for-zenhub/67

## Test Plan
- New tests have been added
- Make sure all existing tests pass
- 
Screenshot 2021-11-26 at 6 15 03 PM

## Documentation changes
NA

## Checklist for important updates
- [ ] ~~Changelog has been updated~~
- [ ] ~~`frontendDriverInterfaceSupported.json` file has been updated (if needed)~~
   - ~~Along with the associated array in `lib/ts/version.ts`~~
- [ ] ~~Changes to the version if needed~~
   - ~~In `package.json`~~
   - ~~In `package-lock.json`~~
   - ~~In `lib/ts/version.ts`~~
- [x] Had run `npm run build-pretty`
- [x] Had installed and ran the pre-commit hook
- [x] Issue this PR against the latest non released version branch.
   - To know which one it is, run find the latest released tag (`git tag`) in the format `vX.Y.Z`, and then find the latest branch (`git branch --all`) whose `X.Y` is greater than the latest released tag.
   - If no such branch exists, then create one from the latest released branch.

## Remaining TODOs for this PR
- [ ] Test that APIs for JWKS URL and open id discovery ones have Access-Control-Allow-Origin as * in their response

### CODE REVIEW

File: test/axios.test.js [hunk 1]
```diff
@@ -709,7 +712,10 @@ describe("Axios AuthHttpRequest class tests", function() {
                 assertEqual(userId, loginResponse.data);
 
                 let data = await supertokens.getAccessTokenPayloadSecurely();
-                assertEqual(Object.keys(data).length, 0);
+
+                if (!jwtEnabled) {
+                    assertEqual(Object.keys(data).length, 0);
+                }
```
Comment (rishabhpoddar) [inline test/axios.test.js:718]: Else it should be length 2?

File: package.json [hunk 2]
```diff
@@ -15,6 +15,7 @@
     "axios": "*",
     "estimo": "^2.2.9",
     "isomorphic-fetch": "2.2.1",
+    "jsonwebtoken": "^8.5.1",
```
Comment (rishabhpoddar) [inline package.json:18]: Should also bump the patch version of our package

File: test/server/index.js [hunk 3]
```diff
@@ -107,6 +157,25 @@ app.post("/startST", async (req, res) => {
     res.send(pid + "");
 });
 
+app.get("/featureFlags", async (req, res) => {
+    res.status(200).json({
+        sessionJwt: maxVersion(supertokens_node_version, "8.3") === supertokens_node_version
+    });
+});
+
+app.post("/resetST", async (req, res) => {
```
Comment (rishabhpoddar) [inline test/server/index.js:166]: Call it /reinitialiseBackendConfig

File: test/server/index.js [hunk 4]
```diff
@@ -107,6 +157,25 @@ app.post("/startST", async (req, res) => {
...
+    let currentAntiCSRFSetting = SessionRecipeRaw.getInstanceOrThrowError().config.antiCsrf;
+    let jwtPropertyName = req.body.jwtPropertyName;
+
+    console.log(jwtPropertyName);
```
Comment (rishabhpoddar) [inline test/server/index.js:170]: Remove the extra console log

File: test/axios2.test.js [hunk 5]
```diff
@@ -595,4 +596,393 @@ describe("Axios AuthHttpRequest class tests", function() {
...
+                assertEqual(decodedJWT.iss, "http://0.0.0.0:8080");
+                assertEqual(decodedJWT.customClaim, undefined);
+                assertEqual(decodedJWT.newClaim, "newValue");
+            });
+        } finally {
+            await browser.close();
+        }
+    });
+
+    it("Test that access token payload and JWT are valid after the property name changes and payload is updated after the session is created", async function() {
```
Comment (rishabhpoddar) [inline test/axios2.test.js:728]: Change to "after the session is refreshed"

File: test/axios2.test.js [hunk 6]
```diff
@@ -595,4 +596,393 @@ describe("Axios AuthHttpRequest class tests", function() {
...
+                assertEqual(decodedJWT.iss, "http://0.0.0.0:8080");
+                assertEqual(decodedJWT.customClaim, undefined);
+                assertEqual(decodedJWT.newClaim, "newValue");
+            });
+        } finally {
+            await browser.close();
+        }
+    });
+
+    it("Test that access token payload and JWT are valid after the property name changes and ression is refreshed after the session is created", async function() {
```
Comment (rishabhpoddar) [inline test/axios2.test.js:862]: Change ression to session

File: test/axios2.test.js [hunk 7]
```diff
@@ -595,4 +596,393 @@ describe("Axios AuthHttpRequest class tests", function() {
...
+
+                // Verify new JWT
+                assertEqual(decodedJWT.sub, userId);
+                assertEqual(decodedJWT._jwtPName, undefined);
+                assertEqual(decodedJWT.iss, "http://0.0.0.0:8080");
+                assertEqual(decodedJWT.customClaim, "customValue");
+            });
+        } finally {
+            await browser.close();
+        }
```
Comment (rishabhpoddar) [inline test/axios2.test.js:986]: Add a new test that will:
- Create a new session with a JWT
- Wait for the access token to expire
- Call getAccessTokenPayloadSecurely and get the JWT from it
- Make sure that the fetched JWT is a refreshed one and that the refresh API was called exactly once.

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: test/axios.test.js (hunk 1 above)
Comment (nkshah2) [reply test/axios.test.js]: Done, it checks for 5 keys (`sub`, `iss`, `jwt`, `_jwtPName` and `customClaim`)

File: package.json (hunk 2 above)
Comment (nkshah2) [reply package.json:18]: Done

File: test/server/index.js (hunk 3 above)
Comment (nkshah2) [reply test/server/index.js]: Done

File: test/server/index.js (hunk 4 above)
Comment (nkshah2) [reply test/server/index.js]: Done

File: test/axios2.test.js (hunk 5 above)
Comment (nkshah2) [reply test/axios2.test.js:728]: Removed the last few words, they didnt mean anything and were making the test name confusing

File: test/axios2.test.js (hunk 6 above)
Comment (nkshah2) [reply test/axios2.test.js]: Done

File: test/axios2.test.js (hunk 7 above)
Comment (nkshah2) [reply test/axios2.test.js:986]: Done, added for both fetch and axios

**Pull request merged**